test/test_tools_add_pre_post_processing_to_model.py

```diff
@@ -158,24 +185,189 @@ def test_pytorch_superresolution(self):
+                                 "CPUExecutionProvider"])
+
+        result = s.run(
+            None, {s.get_inputs()[0].name: np.array([[*input_text]])})
+
+        # SentencePieceTokenizer in ORT is round to zero, so we need to use atol=1
+        self.assertEqual(np.allclose(result[0], ref_output[0], atol=1), True)
+        self.assertEqual(np.allclose(result[1], ref_output[1]), True)
+
+    def test_bert_tokenizer(self):
+        output_model = (Path(test_data_dir) /
+                        "../bert_tokenizer.onnx").resolve()
+        input_text, ref_output = self.build_testModel_and_get_ref_output_for_tokenizer(
+            "BertTokenizer", output_model)
+
+        so = ort.SessionOptions()
+        so.register_custom_ops_library(get_library_path())
+        s = ort.InferenceSession(str(output_model), so, providers=[
+                                 "CPUExecutionProvider"])
+
+        result = s.run(
+            None, {s.get_inputs()[0].name: np.array([[*input_text]])})
+
+        self.assertEqual(np.allclose(result[0], ref_output[0]), True)
+        self.assertEqual(np.allclose(result[1], ref_output[2]), True)
+        self.assertEqual(np.allclose(result[2], ref_output[1]), True)
+
+    def test_hfBert_tokenizer(self):
+        output_model = (Path(test_data_dir) /
+                        "../hfbert_tokenizer.onnx").resolve()
+        input_text, ref_output = self.build_testModel_and_get_ref_output_for_tokenizer(
+            "HfBertTokenizer", output_model)
+
+        so = ort.SessionOptions()
+        so.register_custom_ops_library(get_library_path())
+        s = ort.InferenceSession(str(output_model), so, providers=[
+                                 "CPUExecutionProvider"])
+
+        result = s.run(None, {s.get_inputs()[0].name: np.array(
+            [[input_text[0], input_text[1]]])})
+
+        self.assertEqual(np.allclose(result[0], ref_output[0]), True)
+        self.assertEqual(np.allclose(result[1], ref_output[2]), True)
+        self.assertEqual(np.allclose(result[2], ref_output[1]), True)
+
+    def test_qatask_with_tokenizer(self):
+        output_model = (Path(test_data_dir) /
+                        "../hfbert_tokenizer.onnx").resolve()
+        input_text, ref_output = self.build_testModel_and_get_ref_output_for_tokenizer(
+            "HfBertTokenizer_with_decoder", output_model
+        )
+
+        so = ort.SessionOptions()
+        so.register_custom_ops_library(get_library_path())
+        s = ort.InferenceSession(str(output_model), so, providers=[
+                                 "CPUExecutionProvider"])
+
+        result = s.run(None, {s.get_inputs()[0].name: np.array(
+            [[input_text[0], input_text[1]]])})
+
+        self.assertEqual(result[0][0], ref_output[0][0])
+
+    # Corner Case
+    def test_debug_step(self):
+        import onnx
+
+        create_named_value = pre_post_processing.utils.create_named_value
+
+        # multiple DebugSteps are stringed together
```

Comment:
nit: this doesn't say what the expected behavior is and it's somewhat hard to infer what we're expecting. we take an existing model, create a new input name but don't add any pre-processing (not sure what happens to that input - is it just a no-op that doesn't become a model output?), and connect up 3 Debug steps with the original model output. 

Can we just add Debug steps in post-processing and say each one should a) pass through a value to the next step and b) create a duplicate output; so for each Debug step we expect 1 additional model output? If so, the check to see whether it works should be based on how many outputs the model originally had and the number of debug steps. 

If that is the case something like 
```
# each Debug step adds a new model output
post_processing = [pre_post_processing.Debug(), pre_post_processing.Debug(), pre_post_processing.Debug()]
...
self.assertEqual(len(new_model.graph.output), len(input_model.graph.output) + len(post_processing))
```

would convey a lot of that information to the developer about the intention of the test.